### PR TITLE
Meter provider sdk

### PR DESCRIPTION
**Pull request by HudsonHumphries**
This PR implements the MeterProvider for the Metrics SDK. This MeterProvider class allows the user to get a Meter and set a controller for the SDK pipeline. Once the user has a meter then they can start recording metric events through various metric instruments. Stubs for the Controller and Meter classes are included so that the code complies and tests pass, these will be implemented in future PRs.

This PR includes tests for GetMeter, asserting that the Meter will not be a null pointer and asserting that all Meters will be equal to each other.

### CODE REVIEW

**Comment by HudsonHumphries** [comment]
We decided that the MeterProvider doesn't need to have the controller in the metrics data path, so I removed it. Rebased as well

File: sdk/include/opentelemetry/sdk/metrics/meter_provider.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+{
+namespace metrics
+{
+class MeterProvider final : public opentelemetry::metrics::MeterProvider
+{
+public:
+  /**
+   * Initialize a new meter provider with a specified controller
+   * @param controller The controller for this meter provider. This must
+   * not be a nullptr.
```
Comment (pyohannes) [inline sdk/include/opentelemetry/sdk/metrics/meter_provider.h:22]: This comment is out of date.

File: sdk/include/opentelemetry/sdk/metrics/meter.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "opentelemetry/sdk/common/atomic_shared_ptr.h"
+#include "opentelemetry/metrics/meter.h"
+#include "opentelemetry/metrics/noop.h"
+#include "opentelemetry/version.h"
```
Comment (pyohannes) [inline sdk/include/opentelemetry/sdk/metrics/meter.h:6]: I think we don't need `atomic_shared_ptr.h` and `noop.h`.

File: sdk/include/opentelemetry/sdk/metrics/meter_provider.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
```
Comment (pyohannes) [inline sdk/include/opentelemetry/sdk/metrics/meter_provider.h:5]: I think we don't (yet) need `map` and `string` includes.

**Comment by HudsonHumphries** [comment]
Thanks for catching those errors! @pyohannes Should be fixed now

**Pull request merged**